gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -1,52 +1,151 @@
-                        .source("text", docContent));
-            }
-
-            // TODO: in practice, you will want to use BulkRequest for speed
-        }
+
+  private static final Logger log = LoggerFactory.getLogger(IndexGigawordWithElasticSearch.class);
+
+  private static final String DEFAULT_INDEX_NAME = "Gigaword";
+
+  private static final String DEFAULT_HOST = "localhost";
+
+  private static final int DEFAULT_PORT_PRI = 9200;
+
+  private static final int DEFAULT_PORT_SEC = 9201;
+
+  private static final String PARAM_FILEPATH = "filePath";
+
+  private static final String PARAM_INDEX_NAME = "indexName";
+
+  private static final String PARAM_HOSTNAME_PRI = "primaryHostName";
+
+  private static final String PARAM_HOSTNAME_SEC = "secondaryHostName";
+
+  private static final String PARAM_PORT_PRI = "primaryPort";
+
+  private static final String PARAM_PORT_SEC = "secondaryPort";
+
+  /**
+   * @param argv takes two arguments, a path to the GZip file and an index name
+   * @throws IOException
+   */
+  public static void main(String[] argv) throws IOException {
+
+    // get parameter file
+    final Parameters parameters;
+
+    if (argv.length > 0) {
+      parameters = Parameters.loadSerifStyle(new File(argv[0]));
+    } else {
+      log.error("Input file required");
+      return;
+    }
+
+    // validate and parse inputs
+    parameters.assertExactlyOneDefined(PARAM_FILEPATH);
+
+    String indexName = parameters.getOptionalString(PARAM_INDEX_NAME).or(DEFAULT_INDEX_NAME);
+    File file = parameters.getExistingFile(PARAM_FILEPATH);
+
+    String contentType = Files.probeContentType(file.toPath());
+    if ((contentType != null && !contentType.equalsIgnoreCase("application/gzip"))
+        || (contentType == null
+            && !FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("gz"))) {
+      log.error("{} is not a valid GZip file", file.getAbsolutePath());
+      return;
     }
+
+    // start indexing
+    try {
+
+      RestHighLevelClient client =
+          new RestHighLevelClient(
+              RestClient.builder(
+                  new HttpHost(
+                      parameters.getOptionalString(PARAM_HOSTNAME_PRI).or(DEFAULT_HOST),
+                      parameters.getOptionalPositiveInteger(PARAM_PORT_PRI).or(DEFAULT_PORT_PRI),
+                      "http"),
+                  new HttpHost(
+                      parameters.getOptionalString(PARAM_HOSTNAME_SEC).or(DEFAULT_HOST),
+                      parameters.getOptionalPositiveInteger(PARAM_PORT_SEC).or(DEFAULT_PORT_SEC),
+                      "http")));
+
+      // indexing
+      index(client, file, indexName);
+      client.close();
+
+    } catch (Exception e) {
+      log.error("Caught exception: {}", e);
+      return;
+    }
+  }
+
+  public static void index(RestHighLevelClient client, File file, String indexName)
+      throws IOException {
+    BulkRequest bulkRequest = new BulkRequest();
```

Comment:
This `BulkRequest` needs to be constructed inside the while loop - otherwise, it just keeps getting bigger and bigger each time.

Reply:
I totally missed it.

Reply:
Fixed, please check.